### PR TITLE
Fix small issues in User Guide

### PR DESCRIPTION
**Pull request by jl-wynen**
This fixes a bunch of small things like typos, broken links, etc.

### CODE REVIEW

File: docs/user-guide/slicing.ipynb [hunk 1]
```diff
@@ -461,11 +461,17 @@
...
     "The first is a fundamental limitation since a slice cannot be defined in such as case.\n",
     "The latter two will likely be supported in the future to some extent."
    ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
```
Comment (SimonHeybrock) [inline docs/user-guide/slicing.ipynb:474]: Empty cell?

**Review by SimonHeybrock — COMMENTED** [review]
Thanks fo the thorough overhaul, a few comments below:

File: docs/user-guide/binned-data/binned-data.ipynb [hunk 2]
```diff
@@ -292,8 +292,8 @@
     "s0 = binned['x', 0]\n",
     "s1 = binned['x', 1]\n",
     "print(f'total events: {sc.sum(binned.bins.sum().data)}')\n",
-    "print(f'events x=0:  {sc.sum(s0.bins.sum().data)}')\n",
-    "print(f'events x=1:  {sc.sum(s1.bins.sum().data)}')"
```
Comment (Matthew-Andrew) [inline docs/user-guide/binned-data/binned-data.ipynb:296]: When I was going through this notebook there seemed to be a couple of matplotlib figures which were created but then not displayed in code cells 5 and 8. Are these figures supposed to be displayed? If so I found adding fig.show() as the final line of the cell resolved things.

Comment (SimonHeybrock) [reply docs/user-guide/binned-data/binned-data.ipynb:296]: I had the problem when running the notebook locally. Are you referring to the docs pages or the notebooks?

Comment (SimonHeybrock) [reply docs/user-guide/binned-data/binned-data.ipynb:296]: Ah, you are right the plots have disappeared. @nvaytet when did this break? Any ideas?

File: docs/user-guide/data-structures.ipynb [hunk 3]
```diff
@@ -169,9 +225,6 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Note that `value` and `variance` include a check ensuring that the data is 0-D.\n",
-    "Using them with, e.g., a 1-D variable with dimension extent 1 will raise an exception.\n",
-    "\n",
```
[the remark this answers is not in the document]
Comment (jl-wynen) [reply docs/user-guide/data-structures.ipynb:174]: Duplicate of https://github.com/scipp/scipp/blob/b808385ee00afa42e8657d8abc0f8d45ce3b1ba7/docs/user-guide/data-structures.ipynb#L255

File: docs/user-guide/slicing.ipynb (hunk 1 above)
Comment (jl-wynen) [reply docs/user-guide/slicing.ipynb:474]: Removed

File: docs/user-guide/binned-data/binned-data.ipynb (hunk 2 above)
Comment (jl-wynen) [reply docs/user-guide/binned-data/binned-data.ipynb:296]: I have the same problem. matplotlib figures are only shown when you use `%matplotlib inline`. So vanilla matplotlib and scipp widgets are incaompatible at the moment.

File: docs/user-guide/groupby.ipynb [hunk 4]
```diff
@@ -102,13 +102,13 @@
    "metadata": {},
    "source": [
     "Apart from `mean`, `groupby` currently also supports `sum` and `flatten`.\n",
```
[the remark this answers is not in the document]
Comment (jl-wynen) [reply docs/user-guide/groupby.ipynb]: Removed sentence comparing it to binning and added links to refs.

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**